src/openrct2/ride/Ride.cpp

```diff
@@ -397,8 +397,23 @@ money32 Ride::CalculateIncomePerHour() const
 
     if (currentShopItem != SHOP_ITEM_NONE)
     {
-        priceMinusCost += price[1];
-        priceMinusCost -= ShopItems[currentShopItem].Cost;
+        money16 photoProfit = price[1] - ShopItems[currentShopItem].Cost;
+
+        if (lifecycle_flags & RIDE_LIFECYCLE_ON_RIDE_PHOTO)
+        {
+            uint32_t photosSold = no_secondary_items_sold;
+            int32_t rideTicketsSold = total_customers - no_secondary_items_sold;
+
+            // for now, ratio is just using all-time values of tickets sold and photos sold instead of last 5 minutes'
+            if (rideTicketsSold > 0)
+            {
+                priceMinusCost = priceMinusCost + ((photosSold * photoProfit) / rideTicketsSold);
```

Comment:
Use a compound operator (+=).

Get rid of the `photosSold` variable, which is not necessary.